### PR TITLE
feat!: Change Accordion `section` boolean prop to `sectionAs` enum

### PR DESCRIPTION
**Pull request by alimpens**

### CODE REVIEW

**Review by RubenSibon — APPROVED** [review]
LGTM

File: packages/react/src/Accordion/Accordion.tsx [hunk 1]
```diff
@@ -28,7 +28,7 @@ const AccordionRoot = forwardRef(
     const { keyDown } = useKeyboardFocus(innerRef, { rotating: true })
 
     return (
-      <AccordionContext.Provider value={{ headingLevel: headingLevel, section: section }}>
+      <AccordionContext.Provider value={{ headingLevel: headingLevel, as: sectionAs }}>
```
Comment (VincentSmedinga) [inline packages/react/src/Accordion/Accordion.tsx:31]: I suggest not renaming at this hierarchical level and only using `as` within `Section`.

File: packages/react/src/Accordion/AccordionContext.tsx [hunk 2]
```diff
@@ -8,12 +8,12 @@ import { HeadingLevel } from '../Heading/Heading'
 
 export type AccordionContextValue = {
   headingLevel: HeadingLevel
-  section?: boolean
+  as?: 'div' | 'section'
```
Comment (VincentSmedinga) [inline packages/react/src/Accordion/AccordionContext.tsx:11]: Can you reorder these and those for the default values?

File: packages/react/src/Accordion/AccordionSection.tsx [hunk 3]
```diff
@@ -46,23 +47,13 @@ export const AccordionSection = forwardRef(
...
-            aria-labelledby={buttonId}
-            className={clsx('ams-accordion__panel', { 'ams-accordion__panel--expanded': isExpanded })}
-          >
-            {children}
-          </div>
-        )}
+        <Tag
+          id={panelId}
+          aria-labelledby={buttonId}
+          className={clsx('ams-accordion__panel', { 'ams-accordion__panel--expanded': isExpanded })}
```
Comment (VincentSmedinga) [inline packages/react/src/Accordion/AccordionSection.tsx:53]: These can be sorted as well.

```suggestion
          aria-labelledby={buttonId}
          className={clsx('ams-accordion__panel', { 'ams-accordion__panel--expanded': isExpanded })}
          id={panelId}
```

File: packages/react/src/Accordion/AccordionContext.tsx (hunk 2 above)
Comment (alimpens) [reply packages/react/src/Accordion/AccordionContext.tsx]: I renamed `as` to `sectionAs`, so it's in the right order again.

**Review by VincentSmedinga — APPROVED** [review]

**Pull request merged**